packages/hir/src/lowering/types.candy

```diff
@@ -43,24 +43,39 @@ fun astTypeToHirType(
             .expand<HirDeclaration>({
               namePart.resolveIdentifier(context, it) as Iterable<HirDeclaration>
             })
-            .toSet() as Set<HirClass | HirTrait | HirProperty | HirFunction | HirModule>)
+            .toSet() as Set<HirType | HirTrait | HirFunction | HirModule>)
         }
 
         // There should only be one candidate left.
```

Comment:
Now that I think about it, one is the maximum, but this could also return no candidates at all, right? If so, we should return a `HirErrorType`.